src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+
+    private final Map<FruitTransaction.Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return handlers.get(operation);
```

getHandler(...) simply returns `handlers.get(operation)` and may return null if `operation` is null or the handler is missing. The description and previous review require that strategy resolution should fail fast with informative errors instead of returning null. This also relates to the checklist item "Follow SOLID rules" (ensure robust abstraction contract). Fix: validate `operation` is not null and throw an informative RuntimeException if the handler is missing, for example: `if (operation == null) throw new RuntimeException("Operation is null"); OperationHandler handler = handlers.get(operation); if (handler == null) throw new RuntimeException("No handler for operation: " + operation); return handler;`.